discovery-provider/src/api/v1/tracks.py

```diff
@@ -288,15 +290,23 @@ def get(self):
         key = self.get_cache_key()
 
         # Attempt to use the cached tracks list
-        full_trending = use_redis_cache(key, TRENDING_TTL_SEC, lambda: get_trending(args))
+        if args['user_id'] is not None:
```

Comment:
don't we want to do this in cache decorator rather than explicitly with trending here? (maybe im wrong)